js/rendererjs/pluginapi.js

```diff
@@ -19,40 +21,43 @@ window.onload = () => {
 	const ReactDOM = require('react-dom')
 	/* eslint-enable global-require */
 
+	const startSiad = () => {
+		Siad.launch(siadConfig)
+	}
 	// Continuously check (every 2000ms) if siad is running.
 	// If siad is not running, disable the plugin by mounting
 	// the `DisabledPlugin` component in the DOM's body.
 	// If siad is running and the plugin has been disabled,
 	// reload the plugin.
-	const checkSiad = () => {
-		Siad.ifRunning(() => {
-			if (disabled) {
-				disabled = false
-				window.location.reload()
-			}
-		}, () => {
-			if (!disabled) {
-				disabled = true
-				ReactDOM.render(<DisabledPlugin startSiad={Siad.start} />, document.body)
-			}
-		})
+	while (true) {
+		const running = await Siad.isRunning(siadConfig.address)
+		if (running && disabled) {
+			disabled = false
+			window.location.reload()
+		}
+		if (!running && !disabled) {
+			disabled = true
+			ReactDOM.render(<DisabledPlugin startSiad={startSiad} />, document.body)
+		}
+		await sleep(2000)
 	}
-
-	checkSiad()
-	// Poll Siad.ifRunning and disable the plugin if siad is not running.
-	setInterval(checkSiad, 2000)
 }
 
-// Siad call wrapper.
-// Only call siad if siad is running.
-const callWrapper = (uri, callback) => {
-	Siad.ifRunning(() => {
-		Siad.call(uri, callback)
-	})
-}
 
 window.SiaAPI = {
-	call: callWrapper,
+	call: async function(url, callback) {
```

Comment:
Not really sure about this function.  Since it's an `async` function, we could return the results instead of calling back, but this breaks compatibility with all plugins using SiaAPI.call.  Maybe it's worth it?